### PR TITLE
Highlight Trash collection in sidebar when viewing root trash collection or a trashed item

### LINKED ISSUE #44805: Trash is not highlighted in the sidebar when opened
### Describe the bug

image

### To Reproduce

1. Open Trash
2. Open sidebar

Trash is not highlighted as selected

### Expected behavior

Trash should be highlighted as selected when open ([like this](https://github.com/metabase/metabase/assets/6830683/7a839b2a-dfb9-45c0-b9de-fb0c603d8c97)).

### Information about your Metabase installation

master, 5260e2a

### Severity

P3

### PR DESCRIPTION
**Pull request by sloansparger**
Closes #44805

### Description

Previously when visiting `/trash` you would not see the trash collection highlighted in the navigation sidebar. This PR makes the adjustment to both highlight the root trash collection and when viewing a trashed item.

### How to verify

Describe the steps to verify that the changes are working as expected.

- Visit root trash collection, verify it's highlighted
- Visit a trash collection/dashboard/question/model, verify that thrash collection is still highlighted 

### Demo

Screenshot 2024-06-14 at 5 22 18 PM
Screenshot 2024-06-14 at 5 22 11 PM
Screenshot 2024-06-14 at 5 21 54 PM
Screenshot 2024-06-14 at 5 21 44 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.tsx [hunk 1]
```diff
@@ -74,6 +78,10 @@ function MainNavbar({
     id: questionId,
   });
 
+  const { data: collection } = useCollectionQuery({
```
Comment (npfitz) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.tsx:81]: Might be a good opportunity to use use `metabase/api`.

File: e2e/test/scenarios/collections/trash.cy.spec.js [hunk 2]
```diff
@@ -624,6 +625,53 @@ describe("scenarios > collections > trash", () => {
...
+    cy.log("Make sure trash is selected for root trash collection");
+    cy.visit("/trash");
+    assertTrashSelectinInNavigationSidebar();
+
+    cy.log("Make sure trash is selected for a trashed collection");
+    cy.get("@collection").then(collection => {
+      cy.intercept("GET", `/api/collection/${collection.id}`).as(
+        "getCollection",
+      );
+      cy.visit(`/collection/${collection.id}-collection-a`);
```
Comment (npfitz) [inline e2e/test/scenarios/collections/trash.cy.spec.js:649]: We have `visitCollection`, `visitQuestion`, and `visitDashboard` helpers in e2e that load the page and wait for various requests to finish. They might be handy here

**Review by npfitz — APPROVED** [review]
Couple of suggestions, but overall it looks good!

**Review by npfitz — APPROVED** [review]

File: e2e/test/scenarios/collections/trash.cy.spec.js [hunk 3]
```diff
@@ -720,6 +769,14 @@ function selectItem(name) {
     .within(() => cy.findByRole("checkbox").click());
 }
 
+function assertTrashSelectinInNavigationSidebar() {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/collections/trash.cy.spec.js:772]:
```suggestion
function assertTrashSelectedInNavigationSidebar() {
```

Comment (sloansparger) [reply e2e/test/scenarios/collections/trash.cy.spec.js:772]: embarrassing 💀

**Pull request merged**